### PR TITLE
Upgrade To Terraform Plugin SDK

### PR DESCRIPTION
**Pull request by rossedman**
This provider is still relying on code from `< 0.12`. To be able to upgrade to newer Terraform versions migrating to `terraform-plugin-sdk` is part of the process and decouples the provider from upstream Terraform. This moves this repo to v1 of the sdk which is backward compatible. See more [here](https://www.terraform.io/docs/extend/guides/v1-upgrade-guide.html)

### CODE REVIEW

**Comment by rossedman** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by chb0github** [comment]
Thanks for the PR - I'm reviewing it

File: pkg/artifactory/provider.go [hunk 1]
```diff
@@ -7,9 +7,8 @@ import (
 
 	artifactoryold "github.com/atlassian/go-artifactory/v2/artifactory"
 	"github.com/atlassian/go-artifactory/v2/artifactory/transport"
-	"github.com/hashicorp/terraform/helper/schema"
-	"github.com/hashicorp/terraform/terraform"
-	"github.com/hashicorp/terraform/version"
```
Comment (rossedman) [inline pkg/artifactory/provider.go:12]: This is deprecated and required some code changes to update. These code changes are standard and recommended by `tf-sdk-migrator`

**Comment by rossedman** [comment]
Here are some instructions for building and running locally as well

---

From your test code directory

```
mkdir -p /path/to/example/code/.terraform/plugins/registry.terraform.io/jfrog/artifactory/2.2.6-alpha/darwin_amd64
```

Then from provider code directory, build and place into your test code directory

```
CGO_ENABLED=0 GOOS=linux go build -o /path/to/example/code//.terraform/plugins/registry.terraform.io/jfrog/artifactory/2.2.6-alpha/darwin_amd64/terraform-provider-artifactory_v2.2.6-alpha
```
Once that is complete declare the new version

```hcl
terraform {
  required_providers {
    artifactory = {
      source  = "jfrog/artifactory"
      version = "2.2.6-alpha"
    }
  }
}
```

**Comment by chb0github** [comment]
tested and looking good

**Review by chb0github — COMMENTED** [review]
Tests pass and I don't have any qualms with this. My only question is: What problem did this solve?

**Comment by rossedman** [comment]
@chb0github This is the recommended way to maintain providers going forward. This provider is out of date and we are using it at my organization and we intend to push more updates to it if we can. This is essentially a `noop` and should provide no breaking changes, but is required in upgrading moving forward. This also decouples the upstream version of terraform in the `go.mod` file to use this SDK instead. See more [here](https://www.terraform.io/docs/extend/guides/v1-upgrade-guide.html)

**Review by chb0github — APPROVED** [review]

**Pull request merged**